crowbar_framework/app/models/node_object.rb

```diff
@@ -1334,6 +1334,22 @@ def boot_device!(device)
   # see barclamp-deployer/chef/cookbooks/barclamp/libraries/barclamp_library.rb
   def unique_name_already_claimed_by(device)
     claimed_name = (crowbar_wall[:claimed_disks] || []).find do |claimed_name, v|
+
+      # If this looks like a virtio disk and the target platform is one
+      # that might not have the "by-path" links (e.g. SLES 12). Avoid
+      # using "by-path".
+      if claimed_name =~ /^\/dev\/vd[a-z]+$/
+        virtio_by_path_platforms = %w(
+          ubuntu-12.04
+          redhat-6.2
+          redhat-6.4
+          centos-6.2
+          centos-6.4
+          suse-11.3
+        )
+        next unless virtio_by_path_platforms.include?(@node[:target_platform])
+      end
+
```

Comment:
Problem is that `crowbar_wall[:claimed_disks]` will return `{"/dev/vda"=>{"owner"=>"Boot"}}` and on SLE12 there is no `by-path`